text/1006-deprecate-action-template-helper.md

```diff
@@ -0,0 +1,205 @@
+---
+stage: accepted
+start-date: 2024-02-13T00:00:00.000Z
+release-date:
+release-versions:
+teams: # delete teams that aren't relevant
+  - cli
+  - data
+  - framework
+  - learning
+  - steering
+  - typescript
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1006
+project-link:
+---
+
+<!---
+Directions for above:
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+-->
+
+# Deprecate `(action)` template helper and `{{action}}` modifier. 
+
+## Summary
+
+The `(action)` template helper and `{{action}}` modifier was common pre-Octane. Now that we have native classes and the `{{on}}` modifier, we no longer need to use `(action)` or `{{action}}`
+
+## Motivation
+
+Remove legacy code with confusing semantics.
+
+This is a part of _[Deprecating Ember Classic (pre-Octane)](https://github.com/emberjs/rfcs/issues/832)_.
+
+## Transition Path
+
+This was written in the [Octave vs Classic cheatsheet](https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/#component-properties__ddau)
+
+<details><summary>that content here</summary>
+
+### Before (pre-Octane)
+
+```js
+// parent-component.js
+import Component from '@ember/component';
+
+export default Component.extend({
+  count: 0
+});
+
+```
+```hbs
+{{!-- parent-component.hbs --}}
+{{child-component count=count}}
+Count: {{this.count}}
+
+```
+```js
+// child-component.js
+import Component from '@ember/component';
+
+export default Component.extend({
+  actions: {
+    plusOne() {
+      this.set('count', this.get('count') + 1);
+    }
+  }
+});
+```
+```hbs
+{{!-- child-component.hbs --}}
+<button type="button" {{action "plusOne"}}>
+  Click Me
+</button>
+```
+
+### After (post-Octane)
+```js
+// parent-component.js
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class ParentComponent extends Component {
+  @tracked count = 0;
+
+  @action plusOne() {
+    this.count++;
+  }
+}
+
+```
+```hbs
+{{!-- parent-component.hbs --}}
+<ChildComponent @plusOne={{this.plusOne}} />
+Count: {{this.count}}
+
+```
+```hbs
+{{!-- child-component.hbs --}}
+<button type="button" {{on "click" @plusOne}}>
+  Click Me
+</button>
+
+```
+
+</details>
+
+But what we could put in the deprecation app:
+
+### Scenario: `action` is passed a string
+
+Before:
+```hbs
+<button type="button" {{action "plusOne"}}>
+  Click Me
+</button>
+```
+
+After
+
```

Comment:
I think this may need to illustrate that the action needs to move out of the `actions` hash in this case.

Reply:
idk if we can know that the function is in the actions hash (explicitly).

methods decorated with `@action` would end up in there

Reply:
I mean we need to show an example in the deprecation guide that if the method is in the actions hash, it needs to move out of it to be used as `this. plusOne`